apps/test/unit/code-studio/components/progress/LessonLockDialogTest.jsx

```diff
@@ -1,112 +1,188 @@
 import React from 'react';
-import {shallow} from 'enzyme';
+import {mount} from 'enzyme';
 import {expect} from '../../../../util/reconfiguredChai';
 import sinon from 'sinon';
 import {UnconnectedLessonLockDialog as LessonLockDialog} from '@cdo/apps/code-studio/components/progress/LessonLockDialog';
 import {LockStatus} from '@cdo/apps/code-studio/lessonLockRedux';
 
+// This * import allows us to stub out the SectionSelector component
+import * as SectionSelector from '@cdo/apps/code-studio/components/progress/SectionSelector';
+
+const fakeSectionId = 42;
+
 const MINIMUM_PROPS = {
-  isOpen: false,
+  isOpen: true,
   handleClose: () => {},
   initialLockStatus: [],
-  selectedSectionId: null,
+  selectedSectionId: fakeSectionId,
   saving: false,
   saveDialog: () => {}
 };
 
-const fakeSectionId = 1;
+// Helper function to get the list of rows in the student table.
+const getStudentRows = wrapper =>
+  wrapper.find('#ui-test-student-table tbody tr');
+
+// Helper function to verify that the radio buttons in the given student row
+// correctly reflect the expected lock status.
+const verifyRowLockStatus = (row, expectedLockStatus) => {
+  const radioButtons = row.find('input');
+  const lockedSelected = radioButtons.at(0).props().checked;
+  expect(lockedSelected).to.equal(expectedLockStatus === LockStatus.Locked);
+  const editableSelected = radioButtons.at(1).props().checked;
+  expect(editableSelected).to.equal(expectedLockStatus === LockStatus.Editable);
+  const readonlySelected = radioButtons.at(2).props().checked;
+  expect(readonlySelected).to.equal(
+    expectedLockStatus === LockStatus.ReadonlyAnswers
+  );
+};
+
+describe('LessonLockDialog with stubbed section selector', () => {
+  // Stub out <SectionSelector> so we can test without Redux.
+  let sectionSelectorStub;
+  beforeEach(() => {
+    sectionSelectorStub = sinon.stub(SectionSelector, 'default');
+    sectionSelectorStub.callsFake(props => null);
+  });
+  afterEach(() => {
+    sectionSelectorStub.restore();
+  });
+
+  it('renders with minimal props', () => {
+    const wrapper = mount(<LessonLockDialog {...MINIMUM_PROPS} />);
+    expect(wrapper).not.to.be.null;
+    expect(wrapper.text()).not.to.be.empty;
+  });
 
-describe('LessonLockDialog', () => {
-  it('renders with a selected section', () => {
-    const wrapper = shallow(
-      <LessonLockDialog {...MINIMUM_PROPS} selectedSectionId={fakeSectionId} />
+  it('renders no content when closed', () => {
```

Comment:
ahh.. maybe this answers my question from above. We are now testing the opposite case here?

Reply:
Yes!  (see above)